components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/client/DefaultAuthenticationSeqMgtServiceClient.java

```diff
@@ -0,0 +1,114 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.identity.application.mgt.ui.client;
+
+import org.apache.axis2.client.Options;
+import org.apache.axis2.client.ServiceClient;
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.common.model.xsd.DefaultAuthenticationSequence;
+import org.wso2.carbon.identity.application.mgt.defaultsequence.stub.IdentityDefaultSeqManagementServiceDefaultAuthSeqMgtException;
+import org.wso2.carbon.identity.application.mgt.defaultsequence.stub.IdentityDefaultSeqManagementServiceStub;
+import org.wso2.carbon.identity.application.mgt.ui.util.ApplicationMgtUIConstants;
+
+import java.rmi.RemoteException;
+
+/**
+ * Client for DefaultAuthSeqMgtService.
+ */
+public class DefaultAuthenticationSeqMgtServiceClient {
+
+    private IdentityDefaultSeqManagementServiceStub stub;
+    private static final Log log = LogFactory.getLog(DefaultAuthenticationSeqMgtServiceClient.class);
+    private final boolean debugEnabled = log.isDebugEnabled();
+
+    public DefaultAuthenticationSeqMgtServiceClient(String cookie, String backendServerURL,
+                                                    ConfigurationContext configCtx) throws Exception {
+
+        String serviceURL = backendServerURL + "IdentityDefaultSeqManagementService";
+        stub = new IdentityDefaultSeqManagementServiceStub(configCtx, serviceURL);
+
+        ServiceClient client = stub._getServiceClient();
+        Options option = client.getOptions();
+        option.setManageSession(true);
+        option.setProperty(org.apache.axis2.transport.http.HTTPConstants.COOKIE_STRING, cookie);
+
+        if (debugEnabled) {
+            log.debug("Invoking service " + serviceURL);
+        }
+    }
+
+    /**
+     * Create default authentication sequence.
+     *
+     * @param authenticationSequence default authentication sequence
+     * @throws IdentityDefaultSeqManagementServiceDefaultAuthSeqMgtException
+     */
+    public void createDefaultAuthenticationSeq(DefaultAuthenticationSequence authenticationSequence)
+            throws IdentityDefaultSeqManagementServiceDefaultAuthSeqMgtException {
+
+        try {
+            if (debugEnabled) {
+                log.debug("Creating default authentication sequence.");
+            }
+            stub.createDefaultAuthenticationSeq(authenticationSequence);
+        } catch (RemoteException e) {
+            log.error("Error occurred when creating default authentication sequence.", e);
+            throw new IdentityDefaultSeqManagementServiceDefaultAuthSeqMgtException("Server error occurred.");
```

Comment:
The error msg is not that descriptive. It says just server error occurred. Better to include on performing which operation it failed

Reply:
Or the intention is to hide the fact and respond to the client upon the error only

Reply:
The error message passing from this client won't be visible in UI. The intention here is to log the corresponding server error and throw an exception. Appropriate error message is showing in UI catching it, from https://github.com/wso2/carbon-identity-framework/pull/1927/commits/e851d85b69cff3212a27832d6f167e9879da2cd7#diff-b8e20e08168cfce2f21d7bc41ca377dcR89.